### PR TITLE
[DEV] python version check at package import 

### PR DESCRIPTION
**Pull request by eric-haibin-lin**
## Description ##
Throw an exception instead of the following:
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/haibilin/Library/Python/2.7/lib/python/site-packages/gluonnlp/__init__.py", line 24, in <module>
    from . import data
  File "/Users/haibilin/Library/Python/2.7/lib/python/site-packages/gluonnlp/data/__init__.py", line 23, in <module>
    from . import (batchify, candidate_sampler, conll, corpora, dataloader,
  File "/Users/haibilin/Library/Python/2.7/lib/python/site-packages/gluonnlp/data/corpora/__init__.py", line 23, in <module>
    from . import (google_billion_word, large_text_compression_benchmark, wikitext)
  File "/Users/haibilin/Library/Python/2.7/lib/python/site-packages/gluonnlp/data/corpora/google_billion_word.py", line 35, in <module>
    from ...vocab import Vocab
  File "/Users/haibilin/Library/Python/2.7/lib/python/site-packages/gluonnlp/vocab/__init__.py", line 23, in <module>
    from . import subwords, vocab, bert
  File "/Users/haibilin/Library/Python/2.7/lib/python/site-packages/gluonnlp/vocab/vocab.py", line 176
    def __init__(self, counter: Optional[Counter] = None, max_size: Optional[int] = None,
                              ^
SyntaxError: invalid syntax
```

### Essentials ###
- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by mli** [comment]
Job PR-918/1 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-918/1/index.html

**Comment by mli** [comment]
Job PR-918/3 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-918/3/index.html

File: src/gluonnlp/_pyversion.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Python version check"""
+
+import sys
+
+if not (sys.version_info[0] >= 3 and sys.version_info[1] >= 6):
+    PY3STATEMENT = 'GluonNLP project proudly dropped support of Python2.' \
```
Comment (szha) [inline src/gluonnlp/_pyversion.py]: We could also mention the last version of GluonNLP that supports python2 and how to get it.

File: src/gluonnlp/_pyversion.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+                   'The minimal Python requirement is Python 3.6' \
```
Comment (szha) [inline src/gluonnlp/_pyversion.py]: add period and space `. ` at the end.

File: src/gluonnlp/_pyversion.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Python version check"""
+
+import sys
+
+if not (sys.version_info[0] >= 3 and sys.version_info[1] >= 6):
+    PY3STATEMENT = 'GluonNLP project proudly dropped support of Python2.' \
```
Comment (szha) [inline src/gluonnlp/_pyversion.py]: add space after

**Comment by mli** [comment]
Job PR-918/4 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-918/4/index.html

**Comment by mli** [comment]
Job PR-918/5 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-918/5/index.html

**Comment by leezu** [comment]
How about we follow https://stackoverflow.com/a/45362425/2560672 instead, ie. add `python_requires='>=3.6'` to the `setup.py` file?

**Comment by leezu** [comment]
I added the install time check in https://github.com/dmlc/gluon-nlp/pull/927

**Comment by eric-haibin-lin** [comment]
Closed in favor of https://github.com/dmlc/gluon-nlp/pull/927
